src/app/WriteHandler.cpp

```diff
@@ -256,7 +300,19 @@ CHIP_ERROR WriteHandler::ProcessAttributeDataIBs(TLV::TLVReader & aAttributeData
         }
         else
         {
+            if (ShouldReportListWriteEnd(mProcessingAttributePath, mLastAttributeIsList, dataAttributePath))
+            {
+                DeliverListWriteEnd(mProcessingAttributePath.Value(), CHIP_NO_ERROR);
+            }
+
+            if (ShouldReportListWriteBegin(mProcessingAttributePath, dataAttributePath))
+            {
+                DeliverListWriteBegin(dataAttributePath);
+            }
+
+            mLastAttributeIsList = dataAttributePath.IsListOperation();
```

Comment:
I'm leaning to mProcessingAttributePath since we are also using it for checking conflict writes.

"Last" sounds like "already finished" to me, however, we can only tell if it is really processed when we check the next path.

Then I renamed mLastAttributeIsList